src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1737,6 +1756,15 @@ theorem ofInt_add {n} (x y : Int) : BitVec.ofInt n (x + y) =
   apply eq_of_toInt_eq
   simp
 
+@[simp]
+theorem shiftLeft_add_distrib {x y : BitVec w} {n : Nat} :
+    (x + y) <<< n = x <<< n + y <<< n := by
+  induction n
+  case zero =>
+    simp
+  case succ n ih =>
+    simp [ih, toNat_eq, Nat.shiftLeft_eq_mul_pow, ←Nat.add_mul]
```

Comment:
Our style typically mandates a space after the `←`: `← Nat.add_mul`, same for left arrows in do notation.